src/db/db.cpp

```diff
@@ -156,62 +176,121 @@ DB::Result DB::SelectId(std::string_view table, const std::vector<std::string_vi
 
     std::stringstream select_id;
     select_id << "SELECT " << id_field << " FROM " << table << " WHERE " << conds.str() << ";";
-    const auto& res = Exec(select_id.str(), params);
-    if (res.rows.size() != 1 || res.rows[0].size() != 1) {
-        return {.code = Result::ERROR};
+    auto res = Exec(select_id.str(), params);
+    if (!res.Ok()) {
+        return {res.Code(), std::move(res.Error())};
+    }
+
+    if (res.Value().empty()) {
+        return {StatusCode::NOT_FOUND, "No id for this element was found."};
     }
 
-    return {.id = std::stoull(res.rows[0][0])};
+    if (res.Value()[0].size() != 1) {
+        std::cerr << "'" << select_id.str() << "' returned " << res.Value()[0].size()
```

Comment:
I've checked and we should probably use assert() - https://en.cppreference.com/w/cpp/error/assert
Same in other places where you added exit(2)

Reply:
Assert prints result as:  Assertion `((void)fmt::format(kAssertMessage, select_id.str(), res.Value()[0].size()), res.Value()[0].size() != 1)' failed.

I think cerr and exit(2) give us more readable error.

Reply:
Agreed.